blob/blob.go

```diff
@@ -30,19 +30,16 @@ const MaxShareVersion = 127
 // basic stateless checks over it.
 func New(ns namespace.Namespace, blob []byte, shareVersion uint8) *Blob {
 	return &Blob{
-		NamespaceId:      ns.ID,
+		NamespaceId:      ns.ID(),
 		Data:             blob,
 		ShareVersion:     uint32(shareVersion),
-		NamespaceVersion: uint32(ns.Version),
+		NamespaceVersion: uint32(ns.Version()),
```

Comment:
Do you think it make sense to rework blob in this PR too, so it contains new Namespace, rather than NamespaceID and NamespaceVersion as separate fields?

Reply:
I thought about this and am open to discussion. This would break the state machine which I'm hesitant to do without sufficient reason

Reply:
Digging more into it I've found that Blob struct is actually generated from proto. So proper way to make it would be to create local struct, that has fields and methods for it, while making proto as a wire representation only and putting it into separate pkg. Lets leave it as is and create another issue with canonicalise Blob.

Reply:
Yeah I was thinking the same that way you only verify once upon converting to the local type